jacobian_follower/src/jacobian_follower/jacobian_follower.cpp

```diff
@@ -47,120 +102,140 @@ JacobianFollower::JacobianFollower(std::string const robot_namespace, double con
                                                 std::vector<Eigen::Vector4d> const &preferred_tool_orientations,
-                                                moveit_msgs::RobotState const &start_state,
-                                                std::vector<std::vector<Eigen::Vector3d>> const &grippers,
-                                                double const max_velocity_scaling_factor,
-                                                double const max_acceleration_scaling_factor) {
+                                                moveit_msgs::RobotState const &start_state_msg,
+                                                std::vector<PointSequence> const &grippers,
+                                                double max_velocity_scaling_factor,
+                                                double max_acceleration_scaling_factor) {
+  scene_monitor_->lockSceneRead();
+  auto planning_scene = planning_scene::PlanningScene::clone(scene_monitor_->getPlanningScene());
+  scene_monitor_->unlockSceneRead();
+
   robot_state::RobotState robot_start_state(model_);
-  robotStateMsgToRobotState(start_state, robot_start_state);
-  planning_scene_monitor::LockedPlanningSceneRW planning_scene(scene_monitor_);
-  auto const plan_result = plan(planning_scene, group_name, tool_names, preferred_tool_orientations, robot_start_state,
-                                grippers, max_velocity_scaling_factor, max_acceleration_scaling_factor);
+  robotStateMsgToRobotState(start_state_msg, robot_start_state);
+
+  auto traj_command = make_traj_command_from_python_inputs(
+      planning_scene, group_name, tool_names, grippers, robot_start_state, preferred_tool_orientations,
+      max_velocity_scaling_factor, max_acceleration_scaling_factor);
+  auto const plan_result = plan(traj_command);
   moveit_msgs::RobotTrajectory plan_msg;
   plan_result.first.getRobotTrajectoryMsg(plan_msg);
   plan_msg.joint_trajectory.header.frame_id = "robot_root";
   return std::make_pair(plan_msg, plan_result.second);
 }
 
-PlanResult JacobianFollower::plan(std::string const &group_name, std::vector<std::string> const &tool_names,
-                                  std::vector<Eigen::Vector4d> const &preferred_tool_orientations,
-                                  std::vector<std::vector<Eigen::Vector3d>> const &grippers,
-                                  double const max_velocity_scaling_factor,
-                                  double const max_acceleration_scaling_factor) {
-  planning_scene_monitor::LockedPlanningSceneRW planning_scene(scene_monitor_);
-  auto const &start_state = planning_scene->getCurrentState();
-  debugLogState("start state: ", start_state);
-  return plan(planning_scene, group_name, tool_names, preferred_tool_orientations, start_state, grippers,
-              max_velocity_scaling_factor, max_acceleration_scaling_factor);
+PlanResultMsg JacobianFollower::plan_from_scene_and_state(
```

Comment:
Why the name change? The function signature seems to imply the `scene` and `state` are involved.

You might want to make some structs so this function takes a smaller number of arguments

Reply:
good point, I think all these methods can just be called "plan", during development I ran into issues where the functions differed only by return type which is not allowed